tellor_all_languages/test/test_build_status_pass.js

```diff
@@ -0,0 +1,144 @@
+  //Get the amount of values for that timestamp
+  let count = await tellorOracle.getNewValueCountbyRequestId();
+  await tellorOracle.submitValue(requestId, amount);
+  let time = await getTimestampbyRequestIDandIndex(requestId, count.toString());
+  return time.toNumber();
+};
+
+contract("UsingTellor Tests", function (accounts) {
+	let buildStatusCheck;
+	let tellorOracle;
+
+	beforeEach("Setup contract for each test", async function () {
+	tellorOracle = await Tellor.new();
+	buildStatusCheck = await BuildStatusCheck.new(tellorOracle.address);
+	});
+
+	it("Update Price", async function () {
+	
+		// Specify which data (type) is requested from the oracle. (Should become something like 59, a new entry for TruCol)
+		const requestId = 1; 
+
+
+	// -----------------------------------------Helper Functions ----------------------------
+	// Encode a string to a number 
+	// Source: https://stackoverflow.com/questions/14346829/is-there-a-way-to-convert-a-string-to-a-base-10-number-for-encryption
+	function encode(string) {
+		var number = "0x";
+		var length = string.length;
+		for (var i = 0; i < length; i++)
+			number += string.charCodeAt(i).toString(16);
+		return number;
+	}    	
+    	
+    // Function that runs some incoming shell command (not bash)
+	const exec = require('child_process').exec;
+	function os_func() {
+		this.execCommand = function (cmd) {
+		    return new Promise((resolve, reject)=> {
+		       exec(cmd, (error, stdout, stderr) => {
+		         if (error) {
+		            reject(error);
+		            return;
+		        }
+		        resolve(stdout)
+		       });
+		   })
+	   }
+	}
+	var os = new os_func();
+
+	function create_output_dir(dir) {	
+		var fs = require('fs');
+
+		if (!fs.existsSync(dir)){
+			fs.mkdirSync(dir);
+		}
+	}	
+		
+	
+	// -----------------------------------------Specify Tellor Oracles Data Sources ----------------------------
+	// Read whether a travis build has failed or passed
+	const github_username = "v-bosch"
+	const github_repository_name = "sponsor_example"
+	const github_commit = "b822b1cd82ea169bbc1a7faf3d7855eae6538130"
+	
+	
+	// -----------------------------------------Specify Temporary input and output (files)------------------------
+	// Show the contract contains the logic to identify a correct build fail/pass. 
+	// If the build passes, a uint256 of value 2 is expected 
+	// from the contract. Otherwise a uint of value 1 is expected.
+	const expected_sponsor_contract_output = 2
+	
+	// Specify local output location of curled data
+	var test_output_folder = "curled_test_data"
+	var test_type = "build_status"
+	var test_case = "passed"
+	var output_filename = test_output_folder+"/"+test_type+"/"+test_case+"/"+test_type+"_"+test_case+".json"
+	
+	// Empty test output folder before using it
+	var rimraf = require("rimraf"); //npm install rimraf
+	rimraf(test_output_folder, function () { console.log("Removed the old content of the temporary output directory.\n"); });
```

Comment:
We should be able to make this an async function as follows:
```js
async function deleteFolder(folder) {
   return new Promise((resolve, reject) => {
         rimraf(folder, () => resolve());
    });
}
```
Then we can await it instead of hacking around it :)
_didn't test so let me know if it doesn't work as intended_